include/deal.II/matrix_free/fe_point_evaluation.h

```diff
@@ -69,68 +98,159 @@ namespace internal
       }
 
       static void
-      set_gradient(
-        const Tensor<1, dim, Tensor<1, n_components, VectorizedArray<Number>>>
-          &                value,
-        const unsigned int vector_lane,
-        gradient_type &    result)
+      set_gradient(const interface_vectorized_gradient_type &value,
+                   const unsigned int                        vector_lane,
+                   scalar_gradient_type &                    result)
       {
         for (unsigned int i = 0; i < n_components; ++i)
           for (unsigned int d = 0; d < dim; ++d)
             result[i][d] = value[d][i][vector_lane];
       }
 
       static void
-      get_gradient(
-        Tensor<1, dim, Tensor<1, n_components, VectorizedArray<Number>>> &value,
-        const unsigned int   vector_lane,
-        const gradient_type &result)
+      set_gradient(const interface_vectorized_gradient_type &value,
+                   const unsigned int,
+                   vectorized_gradient_type &result)
+      {
+        for (unsigned int i = 0; i < n_components; ++i)
+          for (unsigned int d = 0; d < dim; ++d)
+            result[i][d] = value[d][i];
+      }
```

Comment:
At least not with the help of `internal::VectorizedArrayTrait::get()` because we need to return a `VectorizedArray` in case the `Number` template of `result` is a `VectorizedArray` and a scalar floating point type if it is `ScalarNumber`.